server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java

```diff
@@ -99,6 +104,33 @@ public boolean testMailboxExists(Username username, MailboxName mailboxName) thr
             .block();
     }
 
+    public Mono<Result> clearMailboxContent(Username username, MailboxName mailboxName, ClearMailboxContentTask.Context context) throws UsersRepositoryException, MailboxException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
+        MessageManager messageManager = mailboxManager.getMailbox(MailboxPath.forUser(username, mailboxName.asString()), mailboxSession);
+
+        return Flux.from(messageManager.listMessagesMetadata(MessageRange.all(), mailboxSession))
+            .map(metaData -> metaData.getComposedMessageId().getUid())
+            .map(messageUid -> deleteMessage(messageManager, messageUid, mailboxSession, context))
```

Comment:
deleteMessage is a disguised blocking call, I prefer to wrap it in a Mono.callable scheduled on the Elastic se=cheduler, inside a concatMap